File: examples/nodejs/redis-queue/producer.js
import Redis from 'ioredis';

const redis = new Redis({ host: process.env.REDIS_HOST || 'localhost', port: 6379, password: process.env.REDIS_PASSWORD || 'change_me' });
const queue = process.env.REDIS_QUEUE || 'jobs:email';

await redis.lpush(queue, JSON.stringify({ id: Date.now(), type: 'email', to: 'user@example.com' }));
console.log('[producer] job pushed');
await redis.quit();

File: examples/python/redis-queue/producer.py
import json
import os
import time
import redis

r = redis.Redis(
    host=os.getenv("REDIS_HOST", "localhost"),
    port=6379,
    password=os.getenv("REDIS_PASSWORD", "change_me"),
    decode_responses=True,
)

r.lpush("jobs:email", json.dumps({"id": int(time.time()), "type": "email"}))
print("[producer] pushed")
